src/reaper_osara.cpp

```diff
@@ -4528,6 +4528,28 @@ void cmdReportItemLength(Command* command) {
 	outputMessage(formatLength(start, end, TF_RULER, FT_NO_CACHE, false));
 }
 
+void cmdReportProjectLength(Command* command) {
+	TimeFormat tf;
+	if (lastCommandRepeatCount == 0) {
```

Comment:
If I'm not mistaken, this is code duplicated from cmdReportCursorPosition. That's a strong indicator that we should split this into a utility function; e.g.
```
// Get the time format to be used for a command where pressing once uses the
// primary ruler unit and pressing twice uses the secondary ruler unit.
TimeFormat getPrimaryOrSecondaryTimeFormatForCommand() {
	if (lastCommandRepeatCount == 0) {
		// Use primary ruler unit.
		return TF_RULER;
	}
	if (GetToggleCommandState(42361)) {
...
```